crates/ruff_linter/src/rules/flake8_markupsafe/rules/unsafe_markup_use.rs

```diff
@@ -60,11 +64,24 @@ impl Violation for UnsafeMarkupUse {
 ///
 /// [markupsafe.Markup]: https://markupsafe.palletsprojects.com/en/stable/escaping/#markupsafe.Markup
 pub(crate) fn unsafe_markup_call(checker: &mut Checker, call: &ExprCall) {
+    if checker
+        .settings
+        .flake8_markupsafe
+        .extend_markup_names
+        .is_empty()
+        && !(checker.semantic().seen_module(Modules::MARKUPSAFE)
+            || checker.semantic().seen_module(Modules::FLASK))
+    {
+        return;
+    }
+    if !is_unsafe_call(call) {
+        return;
+    }
     if let Some(name) = checker
```

Comment:
Would something like this work? (equivalent but in the same return-early style for consistency)

```rs

    let Some(qualified_name) = checker
         .semantic()
         .resolve_qualified_name(&call.func) else{
        return;
    };

    if !is_markup_call(&qualified_name, checker.settings) {
        return;
    } 

    checker
        .diagnostics
        .push(Diagnostic::new(UnsafeMarkupUse { qualified_name.to_string() }, call.range()));
```